app/core/Engine.js

```diff
@@ -84,61 +85,77 @@ class Engine {
           openSeaEnabled: false,
         },
       );
-      const networkController = new NetworkController({
+
+      const networkControllerOpts = {
         infuraProjectId: process.env.MM_INFURA_PROJECT_ID || NON_EMPTY,
-        providerConfig: {
-          static: {
-            eth_sendTransaction: async (
-              payload: { params: any[], origin: any },
-              next: any,
-              end: (arg0: undefined, arg1: undefined) => void,
-            ) => {
-              const { TransactionController } = this.context;
-              try {
-                const hash = await (
-                  await TransactionController.addTransaction(
-                    payload.params[0],
-                    payload.origin,
-                    WalletDevice.MM_MOBILE,
-                  )
-                ).result;
-                end(undefined, hash);
-              } catch (error) {
-                end(error);
-              }
-            },
-          },
-          getAccounts: (
-            end: (arg0: null, arg1: any[]) => void,
-            payload: { hostname: string | number },
+        state: initialState.networkController,
+        messenger: this.controllerMessenger.getRestricted({
+          name: 'NetworkController',
+          allowedEvents: [
+            AppConstants.NETWORK_STATE_CHANGE_EVENT,
+            'NetworkController:providerChange',
+          ],
```

Comment:
Nit: I don't think any allowed events are needed here. It should automatically have permission to use any events or actions under its own namespace

```suggestion
          allowedEvents: [],
```

Reply:
Confirmed